CMS-Enterprise/easi-app
# [NOREF] Add SQLFluff linter

# NOREF

## Description

- Adds new [SQLFluff](https://docs.sqlfluff.com/en/stable/) linter to the pre-commit config
- Run linter

This will lint all `.sql` files that don't live in the `/migrations` folder (since we can't edit old migrations, running this linter on that folder would result in that)

> [!NOTE]
> We're using the SQLFluff [core rules](https://docs.sqlfluff.com/en/stable/reference/rules.html#core-rules) by default, but feel free to look at [all the rules](https://docs.sqlfluff.com/en/stable/reference/rules.html) to see if anything else would be useful!

## How to test this change

- Modify a `.sql` file (somewhere in `/pkg` would be easiest) to update something like indentation, casing, or format
- Run `pre-commit run --all sqlfluff-fix`
- Ensure that the file was fixed / modified

> [!IMPORTANT]
> There's a lot of files touched in this PR - exhaustively going over each one in detail probably isn't reasonable, but I recommend looking over each generally to make sure the _types_ of changes are what we're comfortable with (spacing? formatting? indentation? capitalization?)

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
